modules/Behaviour/behaviour_manage_addProcess.php

```diff
@@ -265,10 +264,7 @@
                 header("Location: {$URL}");
             } else {
                 try {
-                    $data = array('gibbonSchoolYearID' => $session->get('gibbonSchoolYearID'), 'gibbonBehaviourID' => $gibbonBehaviourID, 'gibbonPersonID' => $gibbonPersonID);
-                    $sql = "SELECT * FROM gibbonBehaviour JOIN gibbonPerson ON (gibbonBehaviour.gibbonPersonID=gibbonPerson.gibbonPersonID) JOIN gibbonStudentEnrolment ON (gibbonPerson.gibbonPersonID=gibbonStudentEnrolment.gibbonPersonID) JOIN gibbonFormGroup ON (gibbonStudentEnrolment.gibbonFormGroupID=gibbonFormGroup.gibbonFormGroupID) WHERE gibbonFormGroup.gibbonSchoolYearID=:gibbonSchoolYearID AND status='Full' AND (dateStart IS NULL OR dateStart<='".date('Y-m-d')."') AND (dateEnd IS NULL  OR dateEnd>='".date('Y-m-d')."') AND gibbonBehaviourID=:gibbonBehaviourID AND gibbonBehaviour.gibbonPersonID=:gibbonPersonID";
-                    $result = $connection2->prepare($sql);
-                    $result->execute($data);
+                    $result = $container->get(BehaviourGateway::class)->getExistingBehaviourRecordByID($session->get('gibbonSchoolYearID'), $gibbonBehaviourID, $gibbonPersonID);
```

Comment:
This is strange, as gibbonBehaviourID is already attached to a gibbonPersonID and gibbonSchoolYearID. This method could be replaced with the existing `getBehaviourRecordByID`